src/index.ts

```diff
@@ -25,37 +25,12 @@ export function useMappedState<TState, TResult>(
   mapState: (state: TState) => TResult,
 ): TResult {
   const store = useContext(StoreContext);
+  const [derivedState, setDerivedState] = useState(() =>
```

Comment:
this needs to be after the store check (the conditional hooks should be OK since this won't really happen at runtime, and kills the entire render)